### DISCUSSION TITLE
Using to fetch data?

### DISCUSSION THREAD

**Discussion by SimtecConsult** [next-safe-action/next-safe-action#188] [General]
Has anyone used this library to fetch data from the server?

I've set up a few action clients that check for user roles before doing a mutation. 

I was thinking of using the same clients to fetch data and return to the user on page load.

Has anybody else tried anything similar? Is there something I'm missing as to why this might not work?

**Comment by Alexandredc** [comment]
I think this is a good idea. 
In fact I think we need a `next-safe-loader` 😁 using the same logic (middleware, safe loader client etc...)

**Reply by benjick** [reply]
Is there one available yet? I think I will build one after "use cache" is released
